extensions/amp-iframe/0.1/amp-iframe.js

```diff
@@ -163,6 +172,11 @@ export class AmpIframe extends AMP.BaseElement {
 
   /** @private */
   assertPosition_() {
+    if (this.consentContainer_) {
```

Comment:
We should keep `amp-iframe` empty of any logic related to other elements. 
Two suggestions:
1- Is there an issue with using the normal `must provide a placeholder` instead of changing the `assertPosition` logic? Presence of `placeholder` can be enforced by `amp-consent` validation rules (e.g. have a rule that amp-iframe child of amp-consent must have a placeholder)
2- Created a generic list of `messages to forward` in `amp-iframe`. Have `amp-iframe` blindly dispatch an event when it receives one of those messages. For now the list would only contain `external-consent-flow`.

Reply:
I'm not sure about the mandatory placeholder. This is a pop up iframe, it doesn't much sense to require a placeholder for it. 
I can blindly dispatch the event, but it still needs to be laid out to do so. I would still need to check position of the iframe, which requires me to check if the iframe is within an `<amp-consent>` component.

Reply:
let's keep dependencies one-way and only from amp-consent -> amp-iframe. If mandatory placeholder does not make sense, alternatively amp-consent can inject a placeholder ( if non exists ) during its build callback into defendant amp-iframe.

Apologies for pushing this, intertwined dependencies can really make the code complex and as AMP codebase grows we need to watch out for code complexity as much as possible

Reply:
Talked offline. We need a smarter API to tell `<amp-iframe>` whether it is OK to be placed in first viewport. For example, it is sort of a bug for `<amp-iframe>` within `<amp-lightbox>`. Even though, it requires user interaction to open the lightbox, an `<amp-iframe>` within it still requires a placeholder to be laid out. 

For this PR. We agreed that we should always require a placeholder for `<amp-iframe>` within `<amp-consent>`, because these iframes can be prompted without user interaction, and it makes sense to always include a placeholder to prevent blank space.

Reply:
@aghassemi dependency removed. `<amp-iframe>` will only listen to `consent-response` msg, and blindly dispatch event to the window. 

Unfortunately I can't have `<amp-consent>` directly listen to that postMessage, because it requires `<amp-consent>` to know the exact `iframe` to listen to. And there's not API method for that.